libs/core/include/core/runnable.hpp

```diff
@@ -41,7 +41,8 @@ class Runnable
   }
   virtual void Execute() = 0;
 
-  virtual char const *GetId() const = 0;
+  virtual std::string GetId() const    = 0;
+  virtual std::string GetDebug() const = 0;
```

Comment:
How about a default implementation, for runnables that don't need to report any debug?

Reply:
makse sense. I have done this now